service/src/main/kotlin/no/nav/su/se/bakover/service/søknad/SøknadServiceImpl.kt

```diff
@@ -1,25 +1,92 @@
 package no.nav.su.se.bakover.service.søknad
 import arrow.core.Either
 import arrow.core.getOrElse
+import arrow.core.getOrHandle
 import arrow.core.left
 import arrow.core.right
+import no.nav.su.se.bakover.client.dokarkiv.DokArkiv
+import no.nav.su.se.bakover.client.dokarkiv.Journalpost
+import no.nav.su.se.bakover.client.pdf.PdfGenerator
+import no.nav.su.se.bakover.client.person.PersonOppslag
 import no.nav.su.se.bakover.common.Tidspunkt
 import no.nav.su.se.bakover.database.søknad.SøknadRepo
+import no.nav.su.se.bakover.domain.AktørId
 import no.nav.su.se.bakover.domain.NavIdentBruker.Saksbehandler
 import no.nav.su.se.bakover.domain.Sak
+import no.nav.su.se.bakover.domain.SakFactory
 import no.nav.su.se.bakover.domain.Søknad
+import no.nav.su.se.bakover.domain.SøknadInnhold
+import no.nav.su.se.bakover.domain.oppgave.OppgaveClient
+import no.nav.su.se.bakover.domain.oppgave.OppgaveConfig
 import no.nav.su.se.bakover.service.sak.SakService
 import org.slf4j.LoggerFactory
 import java.util.UUID
 
 internal class SøknadServiceImpl(
     private val søknadRepo: SøknadRepo,
-    private val sakService: SakService
+    private val sakService: SakService,
+    private val sakFactory: SakFactory,
+    private val pdfGenerator: PdfGenerator,
+    private val dokArkiv: DokArkiv,
+    private val personOppslag: PersonOppslag,
+    private val oppgaveClient: OppgaveClient
 ) : SøknadService {
-    val log = LoggerFactory.getLogger(this::class.java)
+    private val log = LoggerFactory.getLogger(this::class.java)
 
-    override fun opprettSøknad(sakId: UUID, søknad: Søknad): Søknad {
-        return søknadRepo.opprettSøknad(sakId, søknad)
+    override fun nySøknad(søknadInnhold: SøknadInnhold): Sak {
+        val sak: Sak = sakService.hentSak(søknadInnhold.personopplysninger.fnr).getOrHandle {
+            log.info("Ny søknad: Fant ikke sak for fødselsnummmer. Oppretter ny sak.")
+            sakFactory.nySak(søknadInnhold.personopplysninger.fnr)
+        }
+        val søknad = Søknad(
+            sakId = sak.id,
+            søknadInnhold = søknadInnhold
+        )
+        opprettJournalpostOgOppgave(sak.id, søknad)
```

Comment:
Jeg har nå snudd om rekkefølgen på disse, så dersom vi ikke klarer å opprette journalpost og oppgave, så vil vi ikke lagre noe i basen.

Hva tenker dere?

Reply:
Høres lurt ut!

Reply:
Tror rekkefølgen er en god idé også! Vet ikke om vi skal påbegynne feilhåndteringen/avbruddshåndteringen i denne PRen, men her kan vi jo f.eks. også vurdere om vi *burde* lagre søknaden, men med felter for hhv. "journalpost opprettet" og "oppgave opprettet", og så har vi ihvertfall et utgangspunkt for å løse feilhåndteringen på et eller annet vis.

Reply:
Etter den endringen jeg foreslo om å lagre søknad og sak i en operasjon ble det litt tungvint og gjøre det i den rekkefølgen jeg ønsket, så jeg gjorde det som det var, ca. 

Så kan vi lage en ny PR på å mellomlagre de forskjellige stegene! La inn en TODO på det